librz/analysis/var_global.c

```diff
@@ -56,75 +77,132 @@ RZ_API RZ_OWN bool rz_analysis_var_global_add(RzAnalysis *analysis, RZ_NONNULL R
 	return true;
 }
 
+/**
+ * \brief Free the global variable instance except the flag
+ * 
+ * \param glob Global variable instance
+ * \return void
+ */
+RZ_API void rz_analysis_var_global_free_except_flag(RZ_NONNULL RzAnalysisVarGlobal *glob) {
+	if (!glob) {
+		return;
+	}
+
+	RZ_FREE(glob->name);
+	rz_type_free(glob->type);
+	rz_vector_fini(&glob->constraints);
+	RZ_FREE(glob);
+}
+
 /**
  * \brief Free the global variable instance
  * 
  * \param glob Global variable instance
+ * \param flags flag list of current core instance
  * \return void
  */
-RZ_API void rz_analysis_var_global_free(RzAnalysisVarGlobal *glob) {
+RZ_API void rz_analysis_var_global_free(RZ_NONNULL RzAnalysisVarGlobal *glob, RZ_NONNULL RzFlag *flags) {
 	if (!glob) {
 		return;
 	}
+	rz_return_if_fail(flags);
 
 	RZ_FREE(glob->name);
 	rz_type_free(glob->type);
 	rz_vector_fini(&glob->constraints);
+
+	if (!rz_flag_unset(flags, glob->flag)) {
+		RZ_LOG_ERROR("Failed to unset flag for global variable %s at 0x%" PFMT64x "\n", glob->name, glob->addr);
+	}
+	glob->flag = NULL;
 	RZ_FREE(glob);
 }
 
+/**
+ * \brief Delete and free the global variable
+ * 
+ * \param analysis RzAnalysis
+ * \param glob global variable to be deleted
+ * \param flags flag list of current core instance
+ * \return true if succeed
+ */
+RZ_API bool rz_analysis_var_global_delete(RZ_NONNULL RzAnalysis *analysis, RZ_NONNULL RzAnalysisVarGlobal *glob, RzFlag *flags) {
```

Comment:
IMO RzFlags should be necessary as an argument only to rz_analysis_var_global_new, then it should never be necessary again.

You could store a pointer to `RzFlags` inside the RzAnalysisVarGlobal.

Reply:
That should be another RZ_BORROW field.